packages/dashboard-frontend/src/containers/Loader/Factory/Steps/Apply/Devfile/index.tsx

```diff
@@ -150,25 +173,45 @@ class StepApplyDevfile extends AbstractLoaderStep<Props, State> {
     }
 
     if (devfile === undefined) {
+      if (factoryResolver === undefined) {
+        const _devfile = defaultDevfile;
+        if (_devfile === undefined) {
+          throw new Error('Failed to resolve the default devfile.');
+        }
+        if (_devfile.projects === undefined) {
+          _devfile.projects = [];
+        }
+        if (_devfile.projects.length === 0) {
+          // adds a default project from the source URL
+          const sourceUrl = new URL(factoryParams.sourceUrl);
+          const name = getProjectName(factoryParams.sourceUrl);
+          const origin = sourceUrl.pathname.endsWith('.git')
+            ? `${sourceUrl.origin}${sourceUrl.pathname}`
+            : `${sourceUrl.origin}${sourceUrl.pathname}.git`;
+          _devfile.projects[0] = { git: { remotes: { origin } }, name };
+          // change default name
+          _devfile.metadata.name = name;
+          _devfile.metadata.generateName = name;
+        }
```

Comment:
Don't you mind moving these lines into the the `updateCurrentDevfile` method?

Reply:
Ok. I fixed it.